Added a check for active tool

ErikOstlund opened the pull request:
Package: `lib-classifier`
Fixes #1688 

Describe your changes:
`onPointerDown` method now checks for an active tool before running. Several prettier updates.

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

shaunanoordin commented on packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.js:
```diff
@@ -1,20 +1,24 @@
-`
-
-function InteractionLayer ({
+import cuid from "cuid";
```
Style guide: `prettier` is adding semicolons to a lot of these lines; these aren't required.

shaunanoordin commented on packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.js:
```diff
@@ -27,105 +31,105 @@ function InteractionLayer ({
-    setCreating(true)
-    activeMark.setSubTaskVisibility(false)
-    return false
+    console.log("activeTool: ", activeTool);
```
`console.log()` needs to be removed

shaunanoordin commented on packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.spec.js:
```diff
@@ -1,31 +1,31 @@
+import DrawingTask from "@plugins/tasks/DrawingTask";
+import { Line, Point } from "@plugins/drawingTools/components";
+
+describe.only("Component > InteractionLayer", function() {
```
This should be `describe()`

shaunanoordin commented on packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.js:
```diff
@@ -27,105 +31,105 @@ function InteractionLayer ({
-    return false
+    console.log("activeTool: ", activeTool);
+
+    if (activeTool.type) {
+      const activeMark = activeTool.createMark({
+        id: cuid(),
+        frame,
+        toolIndex: activeToolIndex
+      });
+      activeMark.initialPosition(convertEvent(event));
+      setActiveMark(activeMark);
+      setCreating(true);
+      activeMark.setSubTaskVisibility(false);
+      return false;
+    }
```
This if-block works great in making sure invalid `activeTool` values don't screw up the app.

However, an easier alternative might be to use `if (!activeTool.type) return false` and not have an if-block

shaunanoordin commented on packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.js:
```diff
@@ -27,105 +31,105 @@ function InteractionLayer ({
+      activeMark.setSubTaskVisibility(false);
+      return false;
+    }
+    return;
```
Recommend: if we're using an if-block, this should be `return false` to be consistent with original code

ErikOstlund commented:
Reverted original commit because my global prettier rules were adding semicolons everywhere and made a huge mess!
In the latest commit,
prettier is now following Standard JS style guide rules.
Removed the console.log
Updated `describe.only()` to `describe()`
Updated `activeTool.type` check as recommended for code clarity and consistency.
All unit tests are passing.
